### PR TITLE
QOLOE-120 Enhancements

### PR DESCRIPTION
**Pull request by MitchAnderson93**
**Current state**
-------------------------------

Dont merge - using as delta

Please review against:
https://ssq-qol.atlassian.net/browse/QOLOE-120 as per enhancement discussion in comments.

**Contains:**
>- Enable embedding of the new generic search component (removing header one) e.g. {{{searchInput}}} 
>- Extending custom 'data-funnelback-profile' custom properties (for script to use)
>- Broke script and styles out into ./searchInput/ component 
>- Tokens integration (into Bootstraps theme-colours via merge) navbar/header plus all project colours changed)
>- Refractor of shared function (funnelback search pattern)
>- Minor style gaps fixed (CTA icons, home icon mobile colour change, hover on mobile changes)
>- Changes to JSON structure 

**Adding:**
>- Add interaction tests for search behaviour (storybook)

**Reminders**
-------------------------------

**Known defects:**
> - Scroll overlay on large site (background should fix)
> - Missing animations (desktop drop down/mobile slide out)
> - Border (none) on final row items in mega menu (desktop) - hence why I made two helper's

**Still to do (non func):**
> - Fix npm audit errors
> - Clean up/review JS/SCSS conventions to improve readability where needed
>- Thorough testing required for mobile menu (popper instance positioning in CSS on mobile/edge cases) 
> - Enhance storybook controls to showcase all element settings (including wrapping each element in light/dark/dark-alt so users have better control of the colour variation settings in metadata

### CODE REVIEW

**Comment by MitchAnderson93** [comment]
> Can we have a default for the search url if not set so others can override if they wish.

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

File: src/components/bs5/searchInput/_colours.scss
Comment (elvishu) [inline src/components/bs5/searchInput/_colours.scss:1]: Nice and tidy, love it.

File: src/components/bs5/navbar/_colours.scss
Comment (elvishu) [inline src/components/bs5/navbar/_colours.scss:1]: Nice clean up.

File: src/components/bs5/header/header.stories.js
Comment (elvishu) [inline src/components/bs5/header/header.stories.js:1]: Great idea importing SearchInput component, and nice clean up on the decorators.

File: src/scss/qld-variables.scss
Comment (elvishu) [inline src/scss/qld-variables.scss:1]: Looks good. It's good idea to change the variable name to $tokens.

File: src/components/bs5/searchInput/searchInput.stories.js
Comment (elvishu) [inline src/components/bs5/searchInput/searchInput.stories.js:1]: Just out curiosity, why the form wrapper <form class="site-search"> is not part of SearchInput component and being added on Storybook rendering only.

**Review by elvishu — APPROVED** [review]
Code looks good. Lots of nice clean up and refactoring.
The colour token update looks ok, although we might need to check if there is other components that are using the replace token - quite unlikely but worth the check.
Good thing is we have Chromatics baseline, so we can do a quick check from there.

**Comment by MitchAnderson93** [comment]
Thanks @elvishu yes colours will need checking, I pointed all the original qld- colours (project colours) to corresponding token value/pair to try to limit breaks where they are used

**Pull request merged**